source/server/connection_handler_impl.h

```diff
@@ -366,87 +135,19 @@ class ConnectionHandlerImpl : public Network::ConnectionHandler,
   UnitFloat listener_reject_fraction_{UnitFloat::min()};
 };
 
-class ActiveUdpListenerBase : public ConnectionHandlerImpl::ActiveListenerImplBase,
-                              public Network::ConnectionHandler::ActiveUdpListener {
-public:
-  ActiveUdpListenerBase(uint32_t worker_index, uint32_t concurrency,
-                        Network::ConnectionHandler& parent, Network::Socket& listen_socket,
-                        Network::UdpListenerPtr&& listener, Network::ListenerConfig* config);
-  ~ActiveUdpListenerBase() override;
-
-  // Network::UdpListenerCallbacks
-  void onData(Network::UdpRecvData&& data) final;
-  uint32_t workerIndex() const final { return worker_index_; }
-  void post(Network::UdpRecvData&& data) final;
-
-  // ActiveListenerImplBase
-  Network::Listener* listener() override { return udp_listener_.get(); }
-
-protected:
-  uint32_t destination(const Network::UdpRecvData& /*data*/) const override {
-    // By default, route to the current worker.
-    return worker_index_;
-  }
-
-  const uint32_t worker_index_;
-  const uint32_t concurrency_;
-  Network::ConnectionHandler& parent_;
-  Network::Socket& listen_socket_;
-  Network::UdpListenerPtr udp_listener_;
-};
-
 /**
- * Wrapper for an active udp listener owned by this handler.
+ * Wrapper for an active listener owned by this handler.
  */
-class ActiveRawUdpListener : public ActiveUdpListenerBase,
-                             public Network::UdpListenerFilterManager,
-                             public Network::UdpReadFilterCallbacks {
+class ActiveListenerImplBase : public virtual Network::ConnectionHandler::ActiveListener {
```

Comment:
Move out of  ConnectionHandlerImpl because ActiveListenerImplBase is used by both Active{Tcp,Udp}Listener. And the latter two classes are moved to separate file